Optimize the manifest in java and python code

xubo245 opened the pull request:
1.support get_sample_list by usage
2.support get inference location
3.support read manifest by java from local and obs
4support write manifest to local and OBS by java code
5.add id and hard 
6.optimize write manifest in python code

gclouding commented on huaweicloud-sdk-python-modelarts-dataset/modelarts/test/test_manifest_write_classification.py:
```diff
@@ -37,26 +37,27 @@ def create_manifest():
                    confidence=annotation_confidence,
                    creation_time=annotation_creation_time,
-                   annotated_by=annotated_by, annotation_format=annotation_format))
+                   annotated_by=annotated_by, annotation_format=annotation_format, property=annotation_property))
```
Consider to rename the parameter name `annotated_by`. Use `annotated_source` ?

xubo245 replied:
It's ok for changing this in this project, but other service need change too. Can we discuss it in a small meeting?

gclouding commented on huaweicloud-sdk-python-modelarts-dataset/modelarts/test/test_manifest_write_classification.py:
```diff
@@ -37,26 +37,27 @@ def create_manifest():
 
 def main(argv):
-  path = os.path.abspath('../') + "/resources/flowers-xy-V201902220937263726_2.manifest"
+  path = os.path.abspath('../../../') + "/resources/classification-xy-V201902220937263726_2.manifest"
```
use os.sep to replace `\` to make the code can run on both linux and windows.

xubo245 replied:
Which system can't work? I test this test class in window 7, it can work successfully!

gclouding commented on huaweicloud-sdk-python-modelarts-dataset/modelarts/test/test_manifest_classification_detection_get_sample_list.py:
```diff
@@ -15,21 +15,34 @@
+  assert sample_list.__len__() == 3
+  for raw_data, label_list in sample_list:
+    assert str(raw_data).startswith("s3://obs-ma/test/classification/datafiles")
+    assert label_list.__len__() == 0
```
why not use `len(label_list)` instead of `label_list.__len__()`

xubo245 replied:
ok, done.  What benefit for using len(label_list) instead of label_list.__len__()?

gclouding commented on huaweicloud-sdk-python-modelarts-dataset/modelarts/test/test_manifest_classification.py:
```diff
@@ -15,18 +15,21 @@
   data_objects = data_set.get_sample_list()
   for data_object in data_objects:
-    assert str(data_object.get_source()).__contains__("s3://obs-ma/test/flowers/datafiles/")
+    assert str(data_object.get_source()).__contains__("s3://obs-ma/test/classification/datafiles/")
```
why not use `in` method instead of `__contains__()`

xubo245 replied:
ok, done. What benefit for using in method instead of __contains__()?

gclouding commented on huaweicloud-sdk-python-modelarts-dataset/modelarts/manifest.py:
```diff
@@ -0,0 +1,371 @@
+    return self.__property
+
+  def get_hard(self):
+    """
```
what does `hard` mean ? Make it self-explanatory ?

xubo245 replied:
hard means this sample is Hard case，which is difficult to handle.

gclouding commented on huaweicloud-sdk-python-modelarts-dataset/modelarts/manifest.py:
```diff
@@ -0,0 +1,371 @@
+
+  def __init__(self, name=None, type=None, loc=None, property=None, confidence=None, creation_time=None,
+               annotated_by=None, annotation_format=None, hard=None):
+    self.__name = name
```
`_` is enough for private variables

xubo245 replied:
ok, done

xubo245 committed:
1.support get_sample_list by usage
    2.support get inference location
    3.add maven framework
    4.support read manifest by Java
    5.support get id in java
    6.Support write and read id in python
    7.Support write manifest to local by java code
    8.Optmize write manifest in java code
    9.write a example for inference result
    10.support set ssl_verify for obs
